### PR TITLE
Change application to exit after receiving an interrupt signal

### PR DESCRIPTION
**Pull request by dpindur**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -171,35 +172,21 @@ func main() {
...
-			break
-		}
-
-		// Don't bother processing commands if a worker has failed and
-		// cannot be recovered
-		if workerErr != nil {
-			running = false
-			break
-		}
+	signalChan := make(chan os.Signal, 1)
```
Comment (dzflack) [inline main.go:175]: Why is this a buffered chan? Is it to prevent multiple routines being spun up when a signal is received?

File: main.go [hunk 2]
```diff
@@ -171,35 +172,21 @@ func main() {
...
+	signal.Notify(signalChan, os.Interrupt)
```
Comment (dzflack) [inline main.go:176]: Why os.Interrupt? does this catch all signals, as opposed to specify syscall.SIGINT, syscall.SIGTERM ?

File: main.go (hunk 1 above)
Comment (dpindur) [reply main.go:175]: If we don't buffer we risk missing the signal as the package does not block while trying to send the signal.
https://godoc.org/github.com/golang/go/src/os/signal#example-Notify

File: main.go (hunk 2 above)
Comment (dpindur) [reply main.go:176]: Notify requires os.Signal of which we only have Interrupt or Kill. I imagine it's a platform agnostic way of implementing it as SIGTERM doesn't exist on Windows.
https://godoc.org/os#Signal

**Pull request merged**